alicloud/resource_alicloud_vpn_route_entry.go

```diff
@@ -0,0 +1,219 @@
+	d.Set("vpn_gateway_id", object.VpnInstanceId)
+
+	if object.State == "published" {
+		d.Set("publish_vpc", true)
+	} else {
+		d.Set("publish_vpc", false)
+	}
+
+	return nil
+}
+
+func resourceAliyunVpnRouteEntryUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*connectivity.AliyunClient)
+	d.Partial(true)
+
+	if d.HasChange("publish_vpc") {
+		request := vpc.CreatePublishVpnRouteEntryRequest()
+		request.RegionId = client.RegionId
+		request.VpnGatewayId = d.Get("vpn_gateway_id").(string)
+		request.RouteDest = d.Get("route_dest").(string)
+		request.NextHop = d.Get("next_hop").(string)
+		request.RouteType = "dbr"
+		request.PublishVpc = requests.NewBoolean(d.Get("publish_vpc").(bool))
+
+		raw, err := client.WithVpcClient(func(vpcClient *vpc.Client) (interface{}, error) {
+			return vpcClient.PublishVpnRouteEntry(request)
+		})
+		if err != nil {
+			return WrapErrorf(err, DefaultErrorMsg, d.Id(), request.GetActionName(), AlibabaCloudSdkGoERROR)
+		}
+		d.SetPartial("public_vpc")
+		addDebug(request.GetActionName(), raw, request.RpcRequest, request)
+	}
+
+	if d.HasChange("weight") {
+		request := vpc.CreateModifyVpnRouteEntryWeightRequest()
+		request.RegionId = client.RegionId
+		request.VpnGatewayId = d.Get("vpn_gateway_id").(string)
+		request.RouteDest = d.Get("route_dest").(string)
+		request.NextHop = d.Get("next_hop").(string)
+		newWeight := d.Get("weight").(int)
+		request.Weight = requests.NewInteger(d.Get("old_weight").(int))
+		request.NewWeight = requests.NewInteger(newWeight)
+
+		raw, err := client.WithVpcClient(func(vpcClient *vpc.Client) (interface{}, error) {
+			return vpcClient.ModifyVpnRouteEntryWeight(request)
+		})
+		if err != nil {
+			return WrapErrorf(err, DefaultErrorMsg, d.Id(), request.GetActionName(), AlibabaCloudSdkGoERROR)
+		}
+		d.SetPartial("weight")
+		d.SetPartial("old_weight")
+		addDebug(request.GetActionName(), raw, request.RpcRequest, request)
+	}
+
+	d.Partial(false)
+	return resourceAliyunVpnRouteEntryRead(d, meta)
+}
+
+func resourceAliyunVpnRouteEntryDelete(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*connectivity.AliyunClient)
+	vpnRouteEntryService := VpnGatewayService{client}
+
+	request := vpc.CreateDeleteVpnRouteEntryRequest()
+	request.RegionId = client.RegionId
+	request.VpnGatewayId = d.Get("vpn_gateway_id").(string)
+	request.RouteDest = d.Get("route_dest").(string)
+	request.NextHop = d.Get("next_hop").(string)
+	request.Weight = requests.NewInteger(d.Get("weight").(int))
+	request.ClientToken = buildClientToken(request.GetActionName())
+
+	err := resource.Retry(15*time.Minute, func() *resource.RetryError {
+		raw, err := client.WithVpcClient(func(vpcClient *vpc.Client) (interface{}, error) {
+			return vpcClient.DeleteVpnRouteEntry(request)
+		})
+		if err != nil {
+			if IsExceptedError(err, VpnConfiguring) {
+				time.Sleep(120 * time.Second)
+				return resource.RetryableError(err)
+			}
+
+			if IsExceptedError(err, UnknownError) {
```

Comment:
If there is a UnknowError, it should not be catched and the specified reason should be found.